compiler/rustc_middle/src/mir/mod.rs

```diff
@@ -2245,7 +2245,7 @@ pub struct Constant<'tcx> {
 #[derive(Lift, TypeFoldable, TypeVisitable)]
 pub enum ConstantKind<'tcx> {
```

Comment:
Isn't the entire point of `mir::ConstantKind` to be the non-shared parts of `mir::Constant`? So if all variants here have a type, shouldn't that be in `mir::Constant`?

Reply:
yea, but that's a bigger refactoring 😆

Reply:
Fair, I guess this can be seen as a first step in that direction. :)

Maybe leave a FIXME in the code or so, just to document that this is not intentional.

Reply:
yea, I have 5 more steps planned before I can confidently remove the `ty` method from `Const`